tests/framework/pvc.go

```diff
@@ -213,79 +199,43 @@ func (f *Framework) VerifyBlankDisk(namespace *k8sv1.Namespace, pvc *k8sv1.Persi
-	if !apierrs.IsAlreadyExists(err) {
-		gomega.Expect(err).ToNot(gomega.HaveOccurred())
-	}
-	err = utils.WaitTimeoutForPodReady(f.K8sClient, executorPod.Name, namespace.Name, utils.PodWaitForTime)
-	gomega.Expect(err).ToNot(gomega.HaveOccurred())
-
-	cmd := fmt.Sprintf("qemu-img info %s/disk.img --output=json", utils.DefaultPvcMountPath)
-
-	output, stderr, err := f.ExecShellInPod(executorPod.Name, namespace.Name, cmd)
-
+	info, err := f.GetImageInfo(namespace, pvc)
 	if err != nil {
-		fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: stderr: [%s]\n", stderr)
 		return false, err
 	}
-	fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: qemu-img info output %s\n", output)
-	var info image.ImgInfo
-	err = json.Unmarshal([]byte(output), &info)
-	if err != nil {
-		klog.Errorf("Invalid JSON:\n%s\n", string(output))
-	}
 	return info.VirtualSize >= info.ActualSize, nil
 }
 
 // VerifyFSOverhead checks whether virtual size is smaller than actual size. That means FS Overhead has been accounted for.
 // NOTE: this assertion is only valid when preallocation is used.
 func (f *Framework) VerifyFSOverhead(namespace *k8sv1.Namespace, pvc *k8sv1.PersistentVolumeClaim, preallocation bool) (bool, error) {
-	var executorPod *k8sv1.Pod
-	var err error
-
 	if !preallocation {
 		return false, fmt.Errorf("VerifyFSOverhead is only valid when preallocation is used")
 	}
 
-	executorPod, err = f.CreateVerifierPodWithPVC(namespace.Name, pvc)
-	if !apierrs.IsAlreadyExists(err) {
-		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	info, err := f.GetImageInfo(namespace, pvc)
+	if err != nil {
+		return false, err
 	}
-	err = utils.WaitTimeoutForPodReady(f.K8sClient, executorPod.Name, namespace.Name, utils.PodWaitForTime)
-	gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
-	cmd := fmt.Sprintf("qemu-img info %s/disk.img --output=json", utils.DefaultPvcMountPath)
-	output, stderr, err := f.ExecShellInPod(executorPod.Name, namespace.Name, cmd)
+	requestedSize := pvc.Spec.Resources.Requests[k8sv1.ResourceStorage]
+	return info.VirtualSize <= info.ActualSize && info.VirtualSize < requestedSize.Value(), nil
+}
 
+// VerifyImagePreallocated checks that image's virtual size is roughly equal to actual size
+func (f *Framework) VerifyImagePreallocated(namespace *k8sv1.Namespace, pvc *k8sv1.PersistentVolumeClaim) (bool, error) {
+	info, err := f.GetImageInfo(namespace, pvc)
 	if err != nil {
-		fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: stderr: [%s]\n", stderr)
 		return false, err
 	}
-	fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: qemu-img info output %s\n", output)
-	var info image.ImgInfo
-	err = json.Unmarshal([]byte(output), &info)
-	if err != nil {
-		klog.Errorf("Invalid JSON:\n%s\n", string(output))
-		return false, nil
-	}
-	requestedSize := pvc.Spec.Resources.Requests[k8sv1.ResourceStorage]
-	return info.VirtualSize <= info.ActualSize && info.VirtualSize < requestedSize.Value(), nil
+
+	return float64(info.VirtualSize)/float64(info.ActualSize) > 0.99, nil
```

Comment:
Please explain this check and use a const for the 0.99

Reply:
Hmm... I think it's wrong. Let me rethink it.

Reply:
OK, the correct check should simply be:
```golang
info.ActualSize >= info.VirtualSize
```
File should take on disk (ActualSize) at least as much, as it reports (VirtualSize). It's `>=` rather than `==` because of file system overhead, block device allocation etc.